com.unity.sg2/Editor/GraphDeltaRegistry/Registry/Registry.cs

```diff
@@ -169,5 +169,13 @@ internal IContextDescriptor GetContextDescriptor(RegistryKey key)
         private IRegistryEntry GetBuilder(RegistryKey key) => builders.TryGetValue(key, out var builder) ? builder : null;
         public static RegistryKey ResolveKey<T>() where T : IRegistryEntry => Activator.CreateInstance<T>().GetRegistryKey();
         public static RegistryFlags ResolveFlags<T>() where T : IRegistryEntry => Activator.CreateInstance<T>().GetRegistryFlags();
+
+        public bool IsLatestVersion(RegistryKey key)
+        {
+            int version = 1;
+            foreach (var versions in builders.Where(kv => kv.Key.Name == key.Name))
+                version = Math.Max(version, versions.Key.Version);
+            return key.Version == version;
+        }
```

Comment:
I added this as a temporary way to prevent old node versions from showing up in the searcher, which is annoyed when you have multiple entries with the same name (since the version doesn't show up in the name).

We need to decide on some UI for showing older versions (eg. do they appear expandable from the current version-
```
v "My Node" (Current Version)
|---Version 2
|---Version 1
```

Reply:
Is there a way we can work around this for the TestUpgrade node? Both versions are needed for the node upgrader manual tests. (Previously each version was put in its own category to prevent the conflict, but if categories aren't working then that won't work.)

Reply:
It fixes itself when the category regression was handled-- so I removed all of this!